lib/ansible/modules/network/aci/aci_switch_leaf_policy_profile.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Bruno Calogero <brunocalogero@hotmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_switch_leaf_policy_profile
+short_description: Create switch policy leaf profiles on Cisco ACI fabrics (infra:NodeP)
+description:
+- Create switch policy leaf profiles on Cisco ACI fabrics.
+- More information from the internal APIC class
+  I(infra:NodeP) at U(https://developer.cisco.com/site/aci/docs/apis/apic-mim-ref/).
+author:
+- Bruno Calogero (@brunocalogero)
+version_added: '2.5'
+options:
+  switch_profile:
+    description:
+    - The name of the Switch Profile.
+    required: yes
+    aliases: [ switch_profile_name ]
+  description:
+    description:
+    - Description for the Switch Profile.
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: aci
+'''
+
+EXAMPLES = r'''
+- name: creating a leaf switch profile
+  aci_switch_leaf_policy_profile:
+    hostname: apic
+    username: someusername
+    password: somepassword
+    switch_profile: sw_name
+    description: sw_description
+    state: present
+'''
+
+RETURN = r'''
+#
+'''
+
+from ansible.module_utils.network.aci.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec
+    argument_spec.update(
+        switch_profile=dict(type='str', aliases=['switch_profile_name']),
```

Comment:
Is this (and the other aliases) actually common in `aci`?
For example `descr` feels like a short form, rather than something that we'd want

Reply:
So we didn't want to alienate the ACI users too much, and prefer to use the same names as they are used to as much as possible. And `descr` is exactly what is used. Only in cases where there could be confusion, we have intervened.

Now in this example, we usually use `name` as the default name parameter of the object we are managing (more in the style of Ansible), but we also allow the user to use a more descriptive name if they prefer to keep key=value pairs equal between subsequent ACI calls, useful if you would be using anchors/references to keep tasks concise.

Again, this is a Work In Progress, so I'd rather let Bruno make the module feature-complete before going into these things.

Reply:
So the short answer, yes we definitely want to add/keep `descr` as an alias :-)

Reply:
@brunocalogero Remember to add `aliases` into the `DESCRIPTION` block

Reply:
It would be nice if the sanity checks could detect inconsistencies in this regard, because these are easy to miss IMO. Same for boolean types and maybe choices.

Reply:
sounds good :) any particular ones you suggest ?

Reply:
`./test/sanity/validate-modules/validate-modules --arg-spec lib/ansible/modules/network/` will spot a few things. Though we don't currently run with `--arg-spec` in CI. We *could* run this when modules are changed